From fc804acc078ef03e2c5b3a233f118a537f260ccd Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Thu, 1 Sep 2022 22:23:19 +0200
Subject: [PATCH] getpcaps: catch PID parsing errors.

Signed-off-by: Jakub Wilk <jwilk@jwilk.net>
Signed-off-by: Andrew G. Morgan <morgan@kernel.org>

conflict:context adaptation
---
 progs/getpcaps.c | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/progs/getpcaps.c b/progs/getpcaps.c
index 8fce0a3..1e914b2 100644
--- a/progs/getpcaps.c
+++ b/progs/getpcaps.c
@@ -39,7 +39,9 @@ int main(int argc, char **argv)
 
     for ( ++argv; --argc > 0; ++argv ) {
 	ssize_t length;
+	long lpid;
 	int pid;
+	char *endarg;
 	cap_t cap_d;
 
 	if (!strcmp(argv[0], "--help") || !strcmp(argv[0], "--usage")) {
@@ -62,7 +64,22 @@ int main(int argc, char **argv)
 	    continue;
 	}
 
-	pid = atoi(argv[0]);
+	errno = 0;
+	lpid = strtol(argv[0], &endarg, 10);
+	if (*endarg != '\0') {
+	    errno = EINVAL;
+	}
+	if (errno == 0) {
+	    if (lpid < 0 || pid != (pid_t) pid)
+		errno = EOVERFLOW;
+	}
+	if (errno != 0) {
+	    fprintf(stderr, "Cannot parse pid %s (%s)\n",
+		    argv[0], strerror(errno));
+	    retval = 1;
+	    continue;
+	}
+	pid = lpid;
 
 	cap_d = cap_get_pid(pid);
 	if (cap_d == NULL) {
-- 
2.27.0

