From 11bdd43001c41d96769e437498bc57e8665ada2f Mon Sep 17 00:00:00 2001
From: zhangchenfeng <zhangchenfeng1@huawei.com>
Date: Fri, 17 Apr 2020 10:21:28 +0800
Subject: [PATCH] libcap-2.61-buildflags

---
 Make.Rules | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Make.Rules b/Make.Rules
index 70d5829..2160012 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -81,10 +81,10 @@ WARNINGS=-Wall -Wwrite-strings -Wpointer-arith -Wcast-qual -Wcast-align \
 	-Wstrict-prototypes -Wmissing-prototypes -Wnested-externs       \
 	-Winline -Wshadow -Wunreachable-code
 COPTS ?= -O2
-CFLAGS ?= $(COPTS) $(DEBUG)
+CFLAGS ?= $(RPM_OPT_FLAGS) $(DEBUG)
 CFLAGS += $(WARNINGS)
 CPPFLAGS += -Dlinux $(DEFINES) $(LIBCAP_INCLUDES)
-LDFLAGS ?= # -g
+LDFLAGS ?= $(RPM_OPT_FLAGS) 
 
 BUILD_CC ?= $(CC)
 BUILD_LD ?= $(BUILD_CC) -Wl,-x -shared
-- 
1.8.3.1

